### PR TITLE
[IMP] hr: Add multi company rule on contract types

### PR DESCRIPTION
**Pull request by tivisse**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_ch/data/account.account.template.csv [hunk 1]
```diff
@@ -141,8 +143,30 @@
...
+"ch_coa_5040","Salary Allowances","5040","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5601","CA Fees","5601","expense","l10n_ch.l10nch_chart_template","False"
 "ch_coa_5700","Social benefits","5700","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5701","Social benefits AC","5701","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5720","Social Benefits LPP","5720","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5721","Social Benefits LPP Redemption","5721","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5730","Social Benefits AANP","5730","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5731","Social Benefits Optional LAAC Redemption","5731","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5740","Social Benefits IJM/CM","5740","expense","l10n_ch.l10nch_chart_template","False"
+"ch_coa_5790","Withholding Taxes IS","5790","expense","l10n_ch.l10nch_chart_template","False"
```
Comment (Levizar) [inline addons/l10n_ch/data/account.account.template.csv:162]: Are we sure `"Withholding Taxes IS"` is an expense account? (and thus by extension, is it the right code?)

From my experience, those are more likely to be of type `liability_current` or `asset_current`.

Also, from what I read here: https://www.taxolution.ch/swiss-tax-guide/withholding-tax-in-switzerland/

It's an expense for the employee not for the company.

I think It's more likely this account will be one of the credited account with when you debit the expense account `"Wages and salaries","5000"`.

**Review by Levizar — COMMENTED** [review]
Hello, as I was pinged, I had a small look at it.
I only have one question for an account in the chart template.

Just by curiosity, where did you get the info for the CH chart of account?

**Comment by tivisse** [comment]
Hi @Levizar 

Thanks for the feedback. This is still a WIP pull request, but I'll keep your comment in mind :)

I got those information from here mainly: https://www.swissdec.ch/fr/versions-et-mises-a-jour/directives-elm/#

File: addons/hr/security/hr_security.xml [hunk 2]
```diff
@@ -86,5 +86,12 @@
         <field name="domain_force">[(1, '=', 1)]</field>
         <field name="groups" eval="[(4, ref('hr.group_hr_user'))]"/>
     </record>
+
+    <record id="ir_rule_hr_contract_type_multi_company" model="ir.rule">
+        <field name="name">HR Contract Type: Multi Company</field>
+        <field name="model_id" ref="model_hr_contract_type"/>
+        <field name="domain_force">['|', ('country_id', '=', False), ('country_id', 'in', user.env.companies.mapped('country_id').ids)]</field>
```
Comment (Bertrand2) [inline addons/hr/security/hr_security.xml:93]: Do we need to map the country_id values?
```suggestion
        <field name="domain_force">['|', ('country_id', '=', False), ('country_id', 'in', user.env.companies.country_id.ids)]</field>
```

**Review by Bertrand2 — COMMENTED** [review]
Just one small question, otherwise LGTM

**Comment by tivisse** [comment]
@robodoo r+ rebase-ff

**Comment by antoine162** [comment]
Hi @tivisse looking at the changes to our Swiss chart of accounts, I'm wondering whether the new accounts you added have been created in sensible places in the CoA?

For example: according to our Swiss Balance Sheet, account codes 290-299 are for Legal Capital Reserves, including Retained Earnings and Current Year Earnings. 

See https://github.com/odoo/enterprise/blob/6417905233ecd6d7f47cb2e2857bba3b6d467fa6/l10n_ch_reports/data/account_financial_html_report_data.xml#L467

To me, `2990 Indemnities` and `2999 Company Car Correction`, don't seem to be Equity accounts at all, and certainly don't have their place with the legal reserves, retained earnings and current year earnings. To me, they should both be Expenses. `2990 Indemnities` should be grouped with the other employee-related expenses in `50xx`. `2999 Company Car Correction` seems to be a depreciation account, so should be grouped with the other depreciation accounts in `68xx`. 

Other accounts for which I have concerns:
- `5003 Salary Advance` should not be of type `expense` but rather of type `asset_prepayment`. It is properly a current asset, since it represents the value of the work the employee is still expected to provide in return for their wages already paid in advance.
- `5790 Withholding Taxes IS` should have a code `220x` like the other Tax Payable accounts. Also, doesn't it fulfil the same function as the already-existing account `2206 Withholding Tax (WT) owed`, so could be removed? https://github.com/odoo/odoo/blob/4694bd1c03a9b41d668e2dbdbb9db8269d8c0683/addons/l10n_ch/data/template/account.account-ch.csv#L78

Once instantiated, it's very difficult to make changes to accounts in customer databases, so it's important to make any fixes before 16.5 is deployed. 

Can you let me know your thoughts regarding these accounts? I can then ask our PO for the Swiss localization (TSB) to review, and open a new PR to make any needed fixes.

**Comment by Levizar** [comment]
Mon mauvais, he asked me and I didn't pay enough attention to that part as the expense part got too much of my attention. :(

**Comment by antoine162** [comment]
> Mon mauvais, he asked me and I didn't pay enough attention to that part as the expense part got too much of my attention. :(

ahhh BIB always the gentleman :smile:

**Pull request closed without merging**